fmn/consumer/worker.py

```diff
@@ -0,0 +1,200 @@
+# FMN worker figuring out for a fedmsg message the list of recipient and
+# contexts
+
+
+import json
+import logging
+import time
+import random
+
+import fmn.lib
+import fmn.rules.utils
+import fedmsg
+import fedmsg.meta
+import fedmsg_meta_fedora_infrastructure
+
+from fedmsg_meta_fedora_infrastructure import fasshim
+
+import pika
+
+import fmn.consumer.fmn_fasshim
+
+log = logging.getLogger("fmn")
+log.setLevel('DEBUG')
+CONFIG = fedmsg.config.load_config()
+fedmsg.meta.make_processors(**CONFIG)
+
+DB_URI = CONFIG.get('fmn.sqlalchemy.uri', None)
+session = fmn.lib.models.init(DB_URI)
+
+from dogpile.cache import make_region
+_cache = make_region(
+    key_mangler=lambda key: "fmn.consumer:dogpile:" + key
+).configure(**CONFIG['fmn.rules.cache'])
+
+valid_paths = fmn.lib.load_rules(root="fmn.rules")
+
+OPTS = pika.ConnectionParameters(
+    heartbeat_interval=0,
+    retry_delay=2,
+)
+
+
+def get_preferences():
+    print 'get_preferences'
+    session = fmn.lib.models.init(DB_URI)
+    prefs = fmn.lib.load_preferences(
+        session, CONFIG, valid_paths,
+        cull_disabled=True,
+        cull_backends=['desktop']
+    )
+    session.close()
+    print 'prefs retrieved'
+    return prefs
+
+
+def update_preferences(openid, prefs):
+    log.info("Loading and caching preferences for %r" % openid)
+    old_preferences = [
+        p for p in prefs if p['user']['openid'] == openid]
+    new_preferences = fmn.lib.load_preferences(
+        session, CONFIG, valid_paths,
+        cull_disabled=True,
+        openid=openid,
+        cull_backends=['desktop']
+    )
+    prefs.extend(new_preferences)
+    for old_preference in old_preferences:
+        prefs.remove(old_preference)
+
+    return prefs
+
+
+CNT = 0
+PREFS = get_preferences()
+
+fmn.consumer.fmn_fasshim.make_fas_cache(**CONFIG)
+# Duck patch fedmsg_meta modules
+fasshim.nick2fas = fmn.consumer.fmn_fasshim.nick2fas
+fasshim.email2fas = fmn.consumer.fmn_fasshim.email2fas
+fedmsg_meta_fedora_infrastructure.supybot.nick2fas = \
+    fmn.consumer.fmn_fasshim.nick2fas
+fedmsg_meta_fedora_infrastructure.anitya.email2fas = \
+    fmn.consumer.fmn_fasshim.email2fas
+fedmsg_meta_fedora_infrastructure.bz.email2fas = \
+    fmn.consumer.fmn_fasshim.email2fas
+fedmsg_meta_fedora_infrastructure.mailman3.email2fas = \
+    fmn.consumer.fmn_fasshim.email2fas
+fedmsg_meta_fedora_infrastructure.pagure.email2fas = \
+    fmn.consumer.fmn_fasshim.email2fas
+
+connection = pika.BlockingConnection(OPTS)
+
+def inform_workers(raw_msg, context, recipients):
+    queue = 'backends'
```

Comment:
Yes, for now I want one queue and I want to monitor it, based on this info we'll split or not